src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -56,11 +58,27 @@
       ;; Respond to all CSR validation failures with a 400
       (middleware-utils/plain-response 400 msg))))
 
+      ;;A recipient MUST ignore If-Modified-Since if the request contains an
+      ;; If-None-Match header field;
```

Comment:
^^ I left these comments here until this can be addressed.  According to the documentation we should ignore this header if "If-None-Match" header is also sent.  I'm unsure if I should implement this logic.  "If-None-Match" header is sent, should I just always send the crl then?  Thoughts?

Reply:
It's hard to say if the "IF" in that comment is the overriding branch, e.g. "If you intend to send this other header, here's how it interacts with 'If-Modified-Since'." Considering we control the only real client for this server, it might be okay as is. But if you do decide to special-case this, I think we should stick to the behavior we have elsewhere of sending the CRL by default.

Reply:
Yeah, if I do add some logic, I think it would just be: if we also get the "If-None-Match" header, then always send the crl (ignore anything in the "If-Modified-Since" header.  Not sure if I need to add anything though.  Any other thoughts?